BM/tools/pcie/pcie_check.c

```diff
@@ -0,0 +1,1225 @@
+		break;
+	case 0x08:
+		printf("PCI/PCI-xto PCi Express Bridge\n");
+		break;
+	case 0x09:
+		printf("Root Complex Integrated Endpoint Device\n");
+		break;
+	case 0x0a:
+		printf("Root Complex Event Collector\n");
+		break;
+	default:
+		printf("reserved\n");
+		break;
+	}
+}
+
+void speedshow(u8 speed)
+{
+	printf("\tspeed: %x   - ", speed);
+	switch (speed) {
+	case 0x00:
+		printf("2.5GT/S");
+		break;
+	case 0x02:
+		printf("5GT/S");
+		break;
+	case 0x04:
+		printf("8GT/S");
+		break;
+	default:
+		printf("reserved");
+		break;
+	}
+	printf("\n");
+}
+
+void linkspeed(u8 speed)
+{
+	printf("\tlink speed:%x   - ", speed);
+	switch (speed) {
+	case 0x01:
+		printf("SupportedLink Speeds Vector filed bit 0");
+		break;
+	case 0x02:
+		printf("SupportedLink Speeds Vector filed bit 1");
+		break;
+	case 0x03:
+		printf("SupportedLink Speeds Vector filed bit 2");
+		break;
+	case 0x04:
+		printf("SupportedLink Speeds Vector filed bit 3");
+		break;
+	case 0x05:
+		printf("SupportedLink Speeds Vector filed bit 4");
+		break;
+	case 0x06:
+		printf("SupportedLink Speeds Vector filed bit 5");
+		break;
+	case 0x07:
+		printf("SupportedLink Speeds Vector filed bit 6");
+		break;
+	default:
+		printf("reserved");
+		break;
+	}
+	printf("\n");
+}
+
+void linkwidth(u8 width)
+{
+	printf("\tlink width:%02x - ", width);
+	switch (width) {
+	case 0x01:
+		printf("x1");
+		break;
+	case 0x02:
+		printf("x2");
+		break;
+	case 0x04:
+		printf("x4");
+		break;
+	case 0x08:
+		printf("x8");
+		break;
+	case 0x0c:
+		printf("x12");
+		break;
+	case 0x10:
+		printf("x16");
+		break;
+	case 0x20:
+		printf("x32");
+		break;
+	default:
+		printf("reserved");
+		break;
+	}
+	printf("\n");
+}
+
+int check_pcie(u32 *ptrdata)
+{
+	u8 ver = 0;
+	u32 next = 0x100, num = 0;
+	u16 offset = 0, cap = 0;
+
+	if (is_pcie == 1) {
+		cap = (u16)(*(ptrdata + next / 4));
+		offset = (u16)(*(ptrdata + next / 4) >> 20);
+		ver = (u8)((*(ptrdata + next / 4) >> 16) & 0xf);
+		// Compile will warning: warning: suggest parentheses around comparison...
+		if ((offset == 0) | (offset == 0xfff)) {
+			printf("PCIE cap:%04x ver:%01x off:%03x|\n", cap, ver, offset);
+			return 0;
+		}
+		printf("PCIE cap:%04x ver:%01x off:%03x|", cap, ver, offset);
+
+		while (1) {
+			num++;
+			cap = (u16)(*(ptrdata + offset / 4));
+			ver = (u8)((*(ptrdata + offset / 4) >> 16) & 0xf);
+			offset = (u16)(*(ptrdata + offset / 4) >> 20);
+
+			if (offset == 0) {
+				printf("cap:%04x ver:%01x off:%03x|\n", cap, ver, offset);
+				break;
+			}
+			printf("cap:%04x ver:%01x off:%03x|", cap, ver, offset);
+			if (num > PCIE_CAP_CHECK_MAX) {
+				printf("PCIE num is more than %d, return\n", PCIE_CAP_CHECK_MAX);
+				break;
+			}
+		}
+	} else {
+		printf("\n");
+	}
+	return 0;
+}
+
+int check_pci(u32 *ptrdata)
+{
+	u8 nextpoint = 0x34;
+	u32 num = 0;
+	u32 *ptrsearch;
+
+	nextpoint = (u8)(*(ptrdata + nextpoint / 4));
+	ptrsearch = ptrdata + nextpoint / 4;
+
+	if ((nextpoint == 0) | (nextpoint == 0xff)) {
```

Comment:
"|" -> "||"

Reply:
It's correct.
For code easy to read, will update it.

Thanks!